lib/chef/provider/package/yum.rb

```diff
@@ -1238,6 +1249,7 @@ def parse_dependency
 
           unless packages.empty?
             new_package_name = packages.first.name
+            new_package_version = packages.first.version.to_s
             Chef::Log.debug("#{@new_resource} no package found for #{@new_resource.package_name} " +
                             "but matched Provides for #{new_package_name}")
```

Comment:
Might be nice for this to include a little more information, but it's just debugging data.

```
debug_msg = "#{@new_resource}: Unable to match package '#{@new_resource.package_name}' but matched #{packages.size} "
debug_msg << package.size == 1 ? "package" : "packages"
debug_msg << ", selected '#{new_package_name}' version '#{new_package_version}'"
Chef::Log.debug(debug_msg)